completion.go

```diff
@@ -0,0 +1,61 @@
+package cli
+
+import (
+	_ "embed"
+	"fmt"
+)
+
+const (
+	completionCommandName = "generate-completion"
+)
+
+//go:embed autocomplete/bash_autocomplete
+var bash_autocomplete string
+
+//go:embed autocomplete/powershell_autocomplete.ps1
+var powershell_autocomplete string
+
+//go:embed autocomplete/zsh_autocomplete
+var zsh_autocomplete string
+
+type renderCompletion func(a *App) (string, error)
+
+func getCompletion(s string) renderCompletion {
+	return func(a *App) (string, error) {
+		return s, nil
+	}
+}
+
+var shellCompletions = map[string]renderCompletion{
+	"bash": getCompletion(bash_autocomplete),
+	"ps":   getCompletion(powershell_autocomplete),
+	"zsh":  getCompletion(zsh_autocomplete),
+	"fish": func(a *App) (string, error) {
+		return a.ToFishCompletion()
+	},
+}
+
+var completionCommand = &Command{
+	Name:   completionCommandName,
+	Hidden: true,
+	Action: func(ctx *Context) error {
+		var shells []string
+		for k := range shellCompletions {
+			shells = append(shells, k)
+		}
```

Comment:
Sort the shells slice for deterministic behavior?

```suggestion
		}
		
		sort.Strings(shells)
```